### PR TITLE
replace old design tokens with the new ones

### PR DESCRIPTION
**Pull request by tishoyanchev**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

Replaced the old design token colors with the new ones. 

<details>
  <summary>📦 Published PR as canary version: <code>2.0.0--canary.42.db62859bc306f44736688524411f64a1b78a6882.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @infineon/design-system-bootstrap@2.0.0--canary.42.db62859bc306f44736688524411f64a1b78a6882.0
  # or 
  yarn add @infineon/design-system-bootstrap@2.0.0--canary.42.db62859bc306f44736688524411f64a1b78a6882.0
  ```
</details>

### CODE REVIEW

File: src/accordion/_accordion.scss [hunk 1]
```diff
@@ -24,15 +24,15 @@
 }
 
 .accordion-button {
-  border: 1px solid tokens.$color-info-200;
-  color: tokens.$color-secondary-500;
+  border: 1px solid tokens.$color-text-white;
```
Comment (nk-radikant) [inline src/accordion/_accordion.scss]: needs to be `tokens.$color-gray-200`

File: src/alert/_alert.scss [hunk 2]
```diff
@@ -111,16 +111,16 @@
 .ifx__alert-icon-wrapper {
   position: relative;
   min-width: 48px;
-  color: tokens.$color-white;
+  color: tokens.$color-bg-white;
```
Comment (nk-radikant) [inline src/alert/_alert.scss]: the semantically correct token is `tokens.$color-text-white`

File: src/button/_button.scss [hunk 3]
```diff
@@ -28,9 +28,9 @@
 
   &:disabled, &.disabled {
     opacity: 1;
-    background-color: tokens.$color-info-300;
-    border-color: tokens.$color-info-300;
-    color: tokens.$color-white;
+    background-color: tokens.$color-gray-300;
+    border-color: tokens.$color-gray-300;
+    color: tokens.$color-bg-white;
```
Comment (nk-radikant) [inline src/button/_button.scss]: the semantically correct token is `tokens.$color-text-white`

File: src/button/_button.scss [hunk 4]
```diff
@@ -40,7 +40,7 @@
   &.btn-warning {
     &.active {
       & .btn-badge {
-        color: tokens.$color-black;
+        color: tokens.$color-bg-darken;
```
Comment (nk-radikant) [inline src/button/_button.scss]: the semantically correct token is `tokens.$color-text-black`

File: src/button/_button.scss [hunk 5]
```diff
@@ -52,126 +52,126 @@
   &.btn-outline-warning {
     &:disabled, &.disabled {
       & .spinner-border {
-        color: tokens.$color-white;
+        color: tokens.$color-bg-white;
```
Comment (nk-radikant) [inline src/button/_button.scss]: the semantically correct token is `tokens.$color-text-white`

File: src/button/_button.scss [hunk 6]
```diff
@@ -52,126 +52,126 @@
...
-      background-color: tokens.$color-primary-600;
-      border-color: tokens.$color-primary-600;  
+      background-color: tokens.$color-default-600;
+      border-color: tokens.$color-default-600;  
     }
 
     &:active, &.active {
-      background-color: tokens.$color-primary-600;
-      border-color: tokens.$color-primary-600;  
+      background-color: tokens.$color-default-600;
```
Comment (nk-radikant) [inline src/button/_button.scss]: needs to be `tokens.$color-default-700`

File: src/button/_button.scss [hunk 7]
```diff
@@ -52,126 +52,126 @@
...
+      border-color: tokens.$color-default-600;
```
Comment (nk-radikant) [inline src/button/_button.scss]: needs to be `tokens.$color-default-700`

File: src/button/_button.scss [hunk 8]
```diff
@@ -52,126 +52,126 @@
...
+      border-color: tokens.$color-default-600;  
     }
   }
 
   &.btn-outline-primary:not(:disabled, .disabled) {
     .btn-badge {
-      background-color: tokens.$color-primary-500;
-      color: tokens.$color-white;
+      background-color: tokens.$color-default-500;
+      color: tokens.$color-bg-white;
```
Comment (nk-radikant) [inline src/button/_button.scss]: the semantically correct token is `tokens.$color-text-white`

File: src/button/_button.scss [hunk 9]
```diff
@@ -52,126 +52,126 @@
...
       & .spinner-border {
-        color: tokens.$color-white;
+        color: tokens.$color-text-white;
       }
     }
 
     &:active, &.active {
-      background-color: tokens.$color-primary-600;
-      border-color: tokens.$color-primary-600;
+      background-color: tokens.$color-default-600;
```
Comment (nk-radikant) [inline src/button/_button.scss]: needs to be `tokens.$color-default-700`

File: src/button/_button.scss [hunk 10]
```diff
@@ -52,126 +52,126 @@
...
+      border-color: tokens.$color-default-600;
```
Comment (nk-radikant) [inline src/button/_button.scss]: needs to be `tokens.$color-default-700`

File: src/button/_button.scss [hunk 11]
```diff
@@ -52,126 +52,126 @@
...
-      background-color: tokens.$color-secondary-600;
-      border-color: tokens.$color-secondary-600;
+      background-color: tokens.$color-highlight-600;
+      border-color: tokens.$color-highlight-600;
     }
 
     &:active, &.active {
-      background-color: tokens.$color-secondary-600;
-      border-color: tokens.$color-secondary-600;  
+      background-color: tokens.$color-highlight-600;
```
Comment (nk-radikant) [inline src/button/_button.scss]: needs to be `tokens.$color-highlight-700`

File: src/button/_button.scss [hunk 12]
```diff
@@ -52,126 +52,126 @@
...
+      border-color: tokens.$color-highlight-600;
```
Comment (nk-radikant) [inline src/button/_button.scss]: needs to be `tokens.$color-highlight-700`

File: src/button/_button.scss [hunk 13]
```diff
@@ -52,126 +52,126 @@
...
       & .spinner-border {
-        color: tokens.$color-white;
+        color: tokens.$color-text-white;
       }
     }
 
     &:active, &.active {
-      background-color: tokens.$color-secondary-600;
-      border-color: tokens.$color-secondary-600;
+      background-color: tokens.$color-highlight-600;
```
Comment (nk-radikant) [inline src/button/_button.scss]: needs to be `tokens.$color-highlight-700`

File: src/button/_button.scss [hunk 14]
```diff
@@ -52,126 +52,126 @@
...
+      border-color: tokens.$color-highlight-600;
```
Comment (nk-radikant) [inline src/button/_button.scss]: needs to be `tokens.$color-highlight-700`

File: src/button/_button.scss [hunk 15]
```diff
@@ -52,126 +52,126 @@
...
 
       & .btn-badge {
-        background-color: tokens.$color-white;
-        color: tokens.$color-black;
+        background-color: tokens.$color-bg-white;
+        color: tokens.$color-bg-darken;
```
Comment (nk-radikant) [inline src/button/_button.scss]: the semantically correct token is `tokens.$color-text-black`

File: src/dropdowns/_dropdown.scss [hunk 16]
```diff
@@ -50,36 +50,36 @@
...
-          color: tokens.$color-black;
+          color: tokens.$color-text-black;
         }
       }
 
       &.active, &:active {
         font-weight: 600;
         background-color: transparent;
-        color: tokens.$color-secondary-500;
+        color: tokens.$color-highlight-500;
```
Comment (nk-radikant) [inline src/dropdowns/_dropdown.scss]: needs to be `tokens.$color-default-500`

File: src/dropdowns/_dropdown.scss [hunk 17]
```diff
@@ -50,36 +50,36 @@
...
 
         & svg {
-          color: tokens.$color-secondary-500;
+          color: tokens.$color-highlight-500;
```
Comment (nk-radikant) [inline src/dropdowns/_dropdown.scss]: needs to be `tokens.$color-default-500`

File: src/form-elements/_form-elements.scss [hunk 18]
```diff
@@ -60,11 +60,11 @@ input[type=range] {
 input[type=range]::-moz-range-thumb {
   width: 14px;
   height: 14px;
-  background-color: tokens.$color-secondary-500;
+  background-color: tokens.$color-highlight-500;
```
Comment (nk-radikant) [inline src/form-elements/_form-elements.scss]: needs to be `tokens.$color-default-500`

File: src/form-elements/_form-elements.scss [hunk 19]
```diff
@@ -60,11 +60,11 @@ input[type=range] {
...
 }
 
 input[type=range]::-moz-range-thumb:active {
-  background-color: tokens.$color-secondary-600;
+  background-color: tokens.$color-highlight-600;
```
Comment (nk-radikant) [inline src/form-elements/_form-elements.scss]: needs to be `tokens.$color-default-700`

File: src/form-elements/_form-elements.scss [hunk 20]
```diff
@@ -73,27 +73,27 @@ input[type=range]:focus::-moz-range-thumb {
...
-  background-color: tokens.$color-info-200;
+  background-color: tokens.$color-gray-200;
   border-radius: 1px;
 }
 
 input[type=range]::-webkit-slider-thumb {
   width: 14px;
   height: 14px;
-  background-color: tokens.$color-secondary-500;
+  background-color: tokens.$color-highlight-500;
```
Comment (nk-radikant) [inline src/form-elements/_form-elements.scss]: needs to be `tokens.$color-default-500`

File: src/button/_button.scss (hunk 6 above)
Comment (tishoyanchev) [reply src/button/_button.scss:72]: Should the border-color remain 600, or should it also be 700?

File: src/form-elements/_form-elements.scss [hunk 21]
```diff
@@ -73,27 +73,27 @@ input[type=range]:focus::-moz-range-thumb {
...
 }
 
 input[type=range]:focus::-webkit-slider-thumb {
   box-shadow: 0 0 0 3px #59B29B66;
 }
 
 input[type=range]::-webkit-slider-thumb:active {
-  background-color: tokens.$color-secondary-600;
+  background-color: tokens.$color-highlight-600;
```
Comment (nk-radikant) [inline src/form-elements/_form-elements.scss]: needs to be `tokens.$color-default-700`

File: src/hero-banner/_hero-banner.scss [hunk 22]
```diff
@@ -78,8 +78,8 @@
 
   &-padded-text {
     @include media-breakpoint-up(lg) {
-      color: tokens.$color-white;
-      background-color: rgba(tokens.$color-secondary-500, .75);
+      color: tokens.$color-text-white;
+      background-color: rgba(tokens.$color-highlight-500, .75);
```
Comment (nk-radikant) [inline src/hero-banner/_hero-banner.scss]: needs to be `tokens.$color-default-500`

File: src/modal/_modal.scss [hunk 23]
```diff
@@ -81,13 +81,13 @@
   }
 
   .ifx__modal-alert-wrapper {
-    background-color: tokens.$color-primary-500;    
+    background-color: tokens.$color-default-500;
```
Comment (nk-radikant) [inline src/modal/_modal.scss]: needs to be `tokens.$color-highlight-500`

**Review by nk-radikant — COMMENTED** [review]
@tishoyanchev  I finished to add comments in the components. I noticed some things which aren’t in the scope of this branch and PR but they still need to be done. I think they are scope of other tickets (and if not, please do these in new branches, so that this branch isn’t to cluttered):

I would add some more comments in the _button.scss, but Github doesn’t allow me to comment on those lines, since there weren’t any changes from you. The changes are: 

Line 188 —> instead of `tokens.$color-success-600` do `tokens.$color-success-700`
Line 189 —> instead of `tokens.$color-success-600` do `tokens.$color-success-700`
Line 220 —> instead of `tokens.$color-success-600` do `tokens.$color-success-700`
Line 221 —> instead of `tokens.$color-success-600` do `tokens.$color-success-700`

Line 246 —> instead of `tokens.$color-danger-600` do `tokens.$color-danger-700`
Line 247 —> instead of `tokens.$color-danger-600` do `tokens.$color-danger-700`
Line 276 —> instead of `tokens.$color-danger-600` do `tokens.$color-danger-700`
Line 277 —> instead of `tokens.$color-danger-600` do `tokens.$color-danger-700`

Line 306 —> instead of `tokens.$color-warning-600` do `tokens.$color-warning-700`
Line 307 —> instead of `tokens.$color-warning-600` do `tokens.$color-warning-700`
Line 338 —> instead of `tokens.$color-warning-600` do `tokens.$color-warning-700`
Line 339 —> instead of `tokens.$color-warning-600` do `tokens.$color-warning-700`


Other than that there are some colours which are used in the templates through classes which also need to be changed:

**ButtonBadgeOnIcon.stories.jsx**
Line 17 —> instead of bg="secondary" do bg="primary"

**Dropdown.jsx**
Line 29 —> instead of variant="secondary" do variant="primary"

**HeroBanner.stories.jsx**
Line 36 —> add color="highlight"

**ListGroup.stories.jsx**
Line 33 —> instead of bg="secondary" do bg="primary"
Line 45 —> instead of bg="secondary" do bg="primary"
Line 51 —> instead of bg="secondary" do bg="primary"

**Modal.stories.jsx**
Line 26 —> instead of color="primary" do color="secondary"
Line 27 —> add color="secondary"

Line 50 —> instead of color="primary" do color="secondary"
Line 51 —> add color=„secondary"

Line 95 —> instead of color="primary" do color="secondary"
Line 96 —> add color=„secondary"


For the alert component the controls variant property values should be changed (primary —> default, secondary —> highlight…)

**Pull request merged**